### PR TITLE
[Android] Implement the API to clear cache for single file.

### PR DESCRIPTION
**Pull request by wuhengzhi**
This patch is to supply an API to clear partial cache.
User could transfer the url to ClearCacheForSingleFile() to clear the
cache for this url.
Add test case for ClearCacheForSingleFile().

BUG=XWALK-4609

### CODE REVIEW

**Comment by wuhengzhi** [comment]
The trybot seems not to be triggered.

**Comment by rakuco** [comment]
This looks like a problem on GitHub's side: according to its status page, it's under another DDoS attack, and in our case it's causing at least the pull request hooks not to be delivered.

**Comment by wuhengzhi** [comment]
@fujunwei Yes, it should be documented.

**Comment by wuhengzhi** [comment]
@lincsoon @halton @axinging PTAL, thanks.
@axinging ```clearCacheForSingleFile()``` should also be wrapped with ```XWalkView.post()```?

**Comment by wuhengzhi** [comment]
ping @lincsoon @halton @axinging, thanks.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 1]
```diff
@@ -511,6 +511,19 @@ public void clearCache(boolean includeDiskFiles) {
     }
 
     /**
+     * Clear the resource cache. Note that it only clear the cache for the specified
+     * url.
+     * @param url indicate which cache will be cleared.
+     * @since 6.0
+     */
+    @XWalkAPI
+    public void clearCacheForSingleFile(String url) {
```
Comment (axinging) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: If there are two XWalkViews, and load the same url, Will all this url related cache being deleted?

Comment (wuhengzhi) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: The answer is Yes since the resurce was cached by url.

**Comment by axinging** [comment]
LGTM

**Comment by lincsoon** [comment]
LGTM

**Comment by HimiGame** [comment]
usr play three website'ex: http://xxx.com/gamex.html',

i want remove one cache .so ,i use clearCacheForSingleFile, like this:

clearCacheForSingleFile('http://xxx.com/gamex.html'); ???

it's not work ~ how i do ,pls help ~ _ ~

btw:
how set cache mode in crosswalk ?

**Pull request merged**